pkg/adapter/awssnssource/adapter.go

```diff
@@ -196,3 +218,20 @@ func healthCheckHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	fmt.Fprintln(w, "OK")
 }
+
+// RegisterHandlerFor implements MTAdapter.
+func (a *adapter) RegisterHandlerFor(ctx context.Context, src *v1alpha1.AWSSNSSource) error {
+	// TODO(antoineco): implement routing
+	return nil
+}
+
+// DeregisterHandlerFor implements MTAdapter.
+func (a *adapter) DeregisterHandlerFor(ctx context.Context, src *v1alpha1.AWSSNSSource) error {
+	// TODO(antoineco): implement routing
+	return nil
+}
+
+// PropagateCondition implements MTAdapter.
+func (a *adapter) PropagateCondition(ctx context.Context, src *v1alpha1.AWSSNSSource, cond *apis.Condition) error {
+	return status.PropagateCondition(ctx, a.statusPatcher, src, cond)
+}
```

Comment:
As mentioned in the PR description, I left the implementation of the HTTP router in TODO and will tackle it in another PR, so this one can focus purely on the interactions between the reconciler and the adapter.